src/components/Form/Location/Address.jsx

```diff
@@ -292,7 +292,7 @@ export default class Address extends ValidationElement {
           </RadioGroup>
         </Show>
 
-        <Show when={this.props.addressBook && book.length}>
+        <Show when={this.props.addressBook && book.length && !this.props.validated}>
```

Comment:
What's the benefit of hiding it once the address is validated? I could see the scenario where someone mistakenly selected the wrong address from the address book and needs to reselect another address.